### PR TITLE
fix undefined check

### LINKED ISSUE #24: Uncaught SyntaxError: Unexpected token u in JSON at position 0
json parse error while providing the following preloadedstate.. 

```
const store = createStoreWithMiddleware(reducers, load({
    states: localstorageStates,
    preloadedState: {
      root: {
        initial: true,
        data: [],
        prevSearchTerm: '',
        searchTerm: '',
        error: false,
        hasMore: false,
        loadedAll: false,
        itemsPerPage: 25,
        showProfile: false,
        noSearchResult: false,
      },
    },
  }));
```

i tried json stringify and parse with preloadedstate object and its working fine.

Thanks

### PR DESCRIPTION
**Pull request by glebtv**
fixes #24

### CODE REVIEW

**Comment by kilkelly** [comment]
This change reintroduces the JSON error for me. I am using Chrome. Which browser are you using? Do you have a repository I can have a look at which contains the error? Thanks.

File: src/index.js [hunk 1]
```diff
@@ -289,7 +289,7 @@ export function load ({
     }
   } else { // Load only specified states into the local Redux state tree
     states.forEach(function (state) {
-      if (localStorage[namespace + '_' + state] !== 'undefined') {
+      if (typeof localStorage[namespace + '_' + state] !== 'undefined') {
```
Comment (Deivis) [inline src/index.js]: A humble suggestion for this validation is to use a positive path first
```suggestion
      if (localStorage[namespace + '_' + state] && localStorage[namespace + '_' + state] !== 'undefined') {
```
This validation verify if the key exists and avoids a an convertion JSON parse error if this key receive the string `undefined`

Comment (kilkelly) [reply src/index.js]: Hi @Deivis , I implemented your pull request locally. Good news, it worked! Better news, it kicked me off to investigate further the cause of the undefined value. I was able to determine that the undefined value shouldn't have been occurring in the first place and squashed that bug once and for all (fingers crossed). I'll explain in more depth in this issue https://github.com/kilkelly/redux-localstorage-simple/issues/24
Thank you!

**Comment by glebtv** [comment]
2.1.6 fixes this for me, so closing this.
Thank you!

**Pull request closed without merging**